### PR TITLE
T22962: Add inline documentation

### PR DESCRIPTION
**Pull request by smspillaz**
https://phabricator.endlessm.com/T22962

### CODE REVIEW

**Comment by smspillaz** [comment]
Reflowed documentation

File: docs/MetadataProvider.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
+# Purpose and Problem Definition
+Previously, the (Companion App
+Service)[https://github.com/endlessm/eos-companion-app-integration] links to
```
Comment (ptomato) [inline docs/MetadataProvider.md]: This section mixes past and present tense.

File: docs/MetadataProvider.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+the Platform SDK and (ekncontent/dmodel)[https://github.com/endlessm/libdmodel]
+directly. This is not a great situation to be in, since it means that we have
+a hard-dependency on apps from a single SDK version. It is against the SDK’s
```
Comment (ptomato) [inline docs/MetadataProvider.md]: You could even say it's against Flatpak's design.

**Comment by smspillaz** [comment]
> The ContentMetadata document reads more like a design document than documentation, but I think that's probably OK for now.

Yeah, I basically imported the design doc directly :p

**Comment by smspillaz** [comment]
Comments addressed

**Pull request merged**